k8s.go

```diff
@@ -0,0 +1,250 @@
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	api "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	defaultK8sNamespace      = "default"
+	defaultK8sResourcePolicy = "keep"
+)
+
+type k8sSecretOptions struct {
+	Namespace       string `long:"namespace" description:"The Kubernetes namespace the secret is located in"`
+	SecretName      string `long:"secret-name" description:"The name of the Kubernetes secret"`
+	SecretEntryName string `long:"secret-entry-name" description:"The name of the entry within the secret"`
+	Base64          bool   `long:"base64" description:"Encode as base64 when storing and decode as base64 when reading"`
+}
+
+func (s *k8sSecretOptions) AnySet() bool {
+	return s.Namespace != defaultK8sNamespace || s.SecretName != "" ||
+		s.SecretEntryName != ""
+}
+
+type helmOptions struct {
+	Annotate       bool   `long:"annotate" description:"Whether Helm annotations should be added to the created secret"`
+	ReleaseName    string `long:"release-name" description:"The value for the meta.helm.sh/release-name annotation"`
+	ResourcePolicy string `long:"resource-policy" description:"The value for the helm.sh/resource-policy annotation"`
+}
```

Comment:
It looks like this is only used to populate specific `labels` and `annotations`, so in the long term it might be preferable to pass in optional lists of each of those instead of limiting them to the these Helm-specific items. Definitely not a blocker though, this can easily be added later.

Reply:
I agree. Will add that in a later PR though, as suggested.